File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        
        n = len(cardPoints)
        length = n - k
        total = sum(cardPoints)
        if length == 0:
            return total
        
        l = 0
        INF = 10 ** 10
        best = INF
        window_sum = 0
        for r in range(n):
            x = cardPoints[r]
            window_sum += x

            if r - l + 1> length:
                window_sum -= cardPoints[l]
                l += 1

            if window_sum < best and r - l + 1 == length:
                best = window_sum
            print(best)

        print(best)
        print(total)
        return total - best

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        
        int n = cardPoints.length;
        int total = 0;

        for (int x : cardPoints) {
            total += x;
        }

        int length = n - k;

        if (length == 0) {
            return total;
        }

        int windowSum = 0;
        for (int r = 0; r < length; r++) {
            windowSum += cardPoints[r];
        }

        int minWindowSum = windowSum;
        int l = 0;
        for (int r = length; r < n; r++) {
            windowSum += cardPoints[r];

            windowSum -= cardPoints[l];
            l++;

            if (windowSum < minWindowSum) {
                minWindowSum = windowSum;
            }
            
        }

        return total - minWindowSum;
    }
}